apps/worker/src/app/workflow/usecases/execute-bridge-job/execute-bridge-job.usecase.ts

```diff
@@ -260,51 +277,62 @@ export class ExecuteBridgeJob {
             transactionId: 1,
           }
         );
-        output = {
-          events: [...digestJobs, job]
-            .map((digestJob) => ({
-              id: digestJob._id,
-              time: digestJob.createdAt,
-              payload: digestJob.payload ?? {},
-            }))
-            .sort((a, b) => new Date(a.time).getTime() - new Date(b.time).getTime()),
-        };
-        break;
+        const events = [...digestJobs, job]
+          .map((digestJob) => ({
+            id: digestJob._id,
+            time: digestJob.createdAt,
+            payload: digestJob.payload ?? {},
+          }))
+          .sort((a, b) => new Date(a.time).getTime() - new Date(b.time).getTime());
+
+        if (isEnhancedDigestEnabled) {
+          return {
+            events,
+            eventCount: events.length,
+          } satisfies DigestResult;
+        }
+
+        return {
+          events,
+        } satisfies DigestResult;
```

Comment:
This is the usecase executed by the worker at the time when we call the bridge. We need to generate the `events, eventCount` values as the step "state", which is later used by Framework client to "compile" the control values with that state.